operator/metadata/updater.go

```diff
@@ -0,0 +1,260 @@
+package metadata
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/attestantio/go-eth2-client/spec/phase0"
+	spectypes "github.com/ssvlabs/ssv-spec/types"
+	"go.uber.org/zap"
+
+	"github.com/ssvlabs/ssv/logging/fields"
+	"github.com/ssvlabs/ssv/protocol/v2/blockchain/beacon"
+	ssvtypes "github.com/ssvlabs/ssv/protocol/v2/types"
+	registrystorage "github.com/ssvlabs/ssv/registry/storage"
+	"github.com/ssvlabs/ssv/storage/basedb"
+)
+
+//go:generate mockgen -package=metadata -destination=./mocks.go -source=./updater.go
+
+const (
+	defaultUpdateInterval    = 12 * time.Minute
+	defaultStreamInterval    = 2 * time.Second
+	defaultUpdateSendTimeout = 30 * time.Second
+	batchSize                = 512
+	streamChanSize           = 1024
+)
+
+type Updater struct {
+	logger            *zap.Logger
+	shareStorage      shareStorage
+	beaconNetwork     beacon.BeaconNetwork
+	fetcher           fetcher
+	updateInterval    time.Duration
+	streamInterval    time.Duration
+	updateSendTimeout time.Duration
+}
```

Comment:
ValidatorMetadata is not so easy to move because storage uses it, and the ValidatorSyncer/Updater uses storage, so there's a circular dependency. I agree we'd better move it but probably later as it needs further refactoring, I don't want to increase the size of this PR as it's already quite big